### PR TITLE
ER ACE experiments

### PR DESCRIPTION
**Pull request by AlbinSou**
I added the Split CIFAR-10 and Split CIFAR-100 experiments for ER-ACE. I get similar results than the one reported for CIFAR100, maybe slightly lower for CIFAR10. 

For 1 seed I get 25.7 % on split cifar100, (25.8 % reported in the paper)
For 1 seed I get 48 % on split cifar10, (52 % reported in the paper)

I have found on my local avalanche branch a change that I can make to the ACECriterion that improves the results a bit more. In particular, right now I am using as current classes the ones that are inferred via the known stream structure (distinct and unique classes in each task). But it works better and makes more sense to use as current classes only the ones that are present in the current batch. Anyway, these changes should add 2~3% on CIFAR10. Getting close from the 52% reported.

In general, I observed that this method is very reliable in that it works pretty consistently across different settings, batch size, optimizer etc.. In comparison to MIR for instance where a slight modification of the learning rate or batch size makes the results move a lot.

### CODE REVIEW

File: experiments/split_cifar10/er_ace.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        batch_size_mem=args.batch_size_mem,
+    )
+
+    ###################
+    #  TRAINING LOOP  #
+    ###################
+
+    print("Starting experiment...")
+
+    print([p.__class__.__name__ for p in cl_strategy.plugins])
```
Comment (AndreaCossu) [inline experiments/split_cifar10/er_ace.py:97]: Is there a reason for this print? Also in the other files.

**Review by AndreaCossu — COMMENTED** [review]
Thanks @AlbinSou , this looks good! 
I left just a minor note on the print command.

**Comment by AndreaCossu** [comment]
TThe stability of the method is actually very relevant to this repo, so very nice! 
Your suggestion on the current classes makes sense to me. If the change does not clutter the code too much (e.g., it does not require you to sub-class strategy etc...) , feel free to go ahead with it. 
Otherwise, you can just add a comment on top of the experiment with that note, so people know about it.

**Comment by AlbinSou** [comment]
> TThe stability of the method is actually very relevant to this repo, so very nice! Your suggestion on the current classes makes sense to me. If the change does not clutter the code too much (e.g., it does not require you to sub-class strategy etc...) , feel free to go ahead with it. Otherwise, you can just add a comment on top of the experiment with that note, so people know about it.

Ok, I think I will make the change later. It actually simplifies the code and the tracking of new vs old classes.

File: experiments/split_cifar10/er_ace.py (hunk 1 above)
Comment (AlbinSou) [reply experiments/split_cifar10/er_ace.py:97]: No, I will remove them, these are just a reflex of mine so that I'm sure that all the plugins that I need are there.

**Comment by AlbinSou** [comment]
Hello @AndreaCossu. I made a few changes, notably I did a PR on avalanche to simplify the ER-ACE criterion, and I also changed a few things here. Like removing input transformations and also removing the validation benchmark that I had introduced. With this we get to better results, however I don't know what's best. Maybe they use validation data in the paper in which case we should add it too. Let me know if you want me to revert to old changes. The results vary a bit when reserving data for validation, but not too much.

Now I have about 50.9% (vs. 52.2 in the paper) for split-cifar10 and 28% (vs. 25.8% in the paper) for split-cifar100.

**Comment by AndreaCossu** [comment]
I think it's best to keep this version without validation data, which does not add much to the experiment flow. Thanks!

**Review by AndreaCossu — APPROVED** [review]
